File: tasty-fast/src/main/java/com/misha/tastyfast/model/CartItem.java
package com.misha.tastyfast.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.math.BigDecimal;

@Getter
@Setter
@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "_cart_item")
@EntityListeners(AuditingEntityListener.class)
public class CartItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer cartItem_id;
    @ManyToOne
    @JoinColumn(name = "cart_id")
    private Cart cart;

    private String itemType;
    private Integer itemId;
    private String itemName;
    private int quantity;
    private BigDecimal price;
}

File: tasty-fast-ui/src/app/services/models/cart-item.ts
/* tslint:disable */
/* eslint-disable */
import { Cart } from '../models/cart';
export interface CartItem {
  cart?: Cart;
  cartItem_id?: number;
  itemId?: number;
  itemName?: string;
  itemType?: string;
  price?: number;
  quantity?: number;
}
